src/Chapter3.hs

```diff
@@ -561,21 +662,36 @@ introducing extra newtypes.
     implementation of the "hitPlayer" function at all!
 -}
 data Player = Player
-    { playerHealth    :: Int
-    , playerArmor     :: Int
-    , playerAttack    :: Int
-    , playerDexterity :: Int
-    , playerStrength  :: Int
+    { playerHealth    :: Health
+    , playerArmor     :: Armor
+    , playerAttack    :: Attack
+    , playerDexterity :: Dexterity
+    , playerStrength  :: Strength
     }
 
-calculatePlayerDamage :: Int -> Int -> Int
-calculatePlayerDamage attack strength = attack + strength
-
-calculatePlayerDefense :: Int -> Int -> Int
-calculatePlayerDefense armor dexterity = armor * dexterity
-
-calculatePlayerHit :: Int -> Int -> Int -> Int
-calculatePlayerHit damage defense health = health + defense - damage
+newtype Attack = MkAttack { attackValue :: Int } deriving (Show)
+newtype Strength = MkStrength { strengthValue :: Int } deriving (Show)
+newtype Damage = MkDamage { damageValue :: Int } deriving (Show)
+newtype Defense = MkDefense { defenseValue :: Int } deriving (Show)
+newtype Health = MkHealth { healthValue :: Int } deriving (Show)
+newtype Armor = MkArmor { armorValue :: Int } deriving (Show)
+newtype Dexterity = MkDexterity { dexterityValue :: Int } deriving (Show)
+
+calculatePlayerDamage :: Attack -> Strength -> Damage
+calculatePlayerDamage attack strength = MkDamage
+  { damageValue = newDamageValue }
+  where
+    newDamageValue = attackValue attack + strengthValue strength
+
+calculatePlayerDefense :: Armor -> Dexterity -> Defense
+calculatePlayerDefense armor dexterity = MkDefense
+  { defenseValue = armorValue armor * dexterityValue dexterity }
+
+calculatePlayerHit :: Damage -> Defense -> Health -> Health
+calculatePlayerHit damage defense health = MkHealth
+  { healthValue = newHealthValue }
+  where
+    newHealthValue = healthValue health + defenseValue defense - damageValue damage
```

Comment:
Nice work on newtypes! :medal_sports: Newtypes are one very important feature in Haskell :+1:
You can read more about the newtype pattern in our blog post: https://kowainik.github.io/posts/haskell-mini-patterns#newtype